src/kubernetes_cluster/spec/kubernetes_api/state_machine.rs

```diff
@@ -54,19 +56,24 @@ pub open spec fn handle_create_request(msg: Message, s: KubernetesAPIState) -> (
         msg.content.is_create_request(),
 {
     let req = msg.content.get_create_request();
-    if s.resources.dom().contains(req.obj.object_ref()) {
-        // Creation fails
+    if req.obj.metadata.namespace.is_Some() && req.namespace != req.obj.metadata.namespace.get_Some_0() {
```

Comment:
But if you don't set namespace for the inner object, when will the state machine go into this branch?

Reply:
If some one sets a wrong namespace in the inner object, it will go to this branch.
In our example controllers we never do so, so we should be good in proof.

Reply:
Note that such error case branches are also used to capture the mistakes if someone sets a wrong namespace in their objects -- if they do so, they will never prove liveness